src/python/antares_xpansion/config_loader.py

```diff
@@ -79,15 +81,22 @@ def _set_simulation_name(self):
             raise ConfigLoader.MissingSimulationName(
                 "Missing argument simulationName")
         elif self._config.simulation_name == "last":
-            self._simulation_name = self._config.simulation_name
+            self._xpansion_simulation_name = self._config.simulation_name
 
         else:
             tmp_zip = Path(self.antares_output()) / \
                 self._config.simulation_name
-            if self.is_zip(tmp_zip):
-                self._last_zip = tmp_zip
-                self._simulation_name = self._last_zip.parent / \
-                    (self._last_zip.stem+"-Xpansion")
+            if self.is_antares_study(tmp_zip):
+                self._last_study = tmp_zip
+                if self.step() in ["resume", "sensitivity"] :
+                    self._xpansion_simulation_name = self._last_study
+                    if self.is_zip(self._last_study):
+                        self._xpansion_simulation_name = self._last_study.parent / self._last_study.stem
+                        with zipfile.ZipFile(self._last_study, 'r') as output_zip:
+                            output_zip.extractall(self._xpansion_simulation_name)
+                else:
+                    self._xpansion_simulation_name = self._last_study.parent / \
+                        (self._last_study.stem+"-Xpansion")
             else:
                 raise ConfigLoader.InvalidSimulationName(
                     f"{tmp_zip} is not a valid zip archive")
```

Comment:
The error message may be misleading as `tmp_zip` is not necessarily a zip archive. Maybe something like `tmp is not a valid Antares output` would be more appropriate.